Private Events  completion

RNtaate opened the pull request:
## What does this PR do?
- [X] Creates User, Event, and Event_attendance models.
- [X] Creates Associations amongst User, Event, and Event_attendance models
- [X] Creates users, events, and sessions controllers.
- [X] Creates simple authentication of users.
- [X] Creates view templates for users, events, and sessions controllers.
- [X] Creates integration tests for project features.

RNtaate committed:
Add starter files

RNtaate committed:
Add rspec and capybara gems

RNtaate committed:
Install rspec

ershadul1 committed:
Add User model

ershadul1 committed:
Add users controller

ershadul1 committed:
Add routes for users controller

ershadul1 committed:
Add view template for user sign up

ershadul1 committed:
Add show template for users controller

RNtaate committed:
Add session controller

RNtaate committed:
Add session routes

RNtaate committed:
Update create method in user controller with session

RNtaate committed:
Design sign up form in new view template

RNtaate committed:
Add validation to user model

RNtaate committed:
Design sign in for session new view template

RNtaate committed:
Design show template for users controller

ershadul1 committed:
Add events model and corresponding attributes

ershadul1 committed:
Complete creator association of event model in user model

ershadul1 committed:
Add events controller

ershadul1 committed:
Add corresponding routing for events controller

ershadul1 committed:
Update show with user created events

ershadul1 committed:
Add index, show view templates for events

RNtaate committed:
Add show view template for events controller

RNtaate committed:
Add index, new, show, create, event_params methods to events controller

RNtaate committed:
Update create, destroy methods in session controller

RNtaate committed:
Add current_user, signed_in?  methods to application controller

RNtaate committed:
Update create method, in users controller

RNtaate committed:
Update events loop in events index view template

ershadul1 committed:
Add new view template for event controller

RNtaate committed:
Update create method to register created events

RNtaate committed:
Add create event link to index view template

RNtaate committed:
Add logout link to view layout.

RNtaate committed:
Create event_attendance model and migrate database

RNtaate committed:
Create associations amongst event_attendance, event, user models

RNtaate committed:
Add events attendees to event show template

RNtaate committed:
Add attended events to user show view template

RNtaate committed:
Add link to user attended_events to events index template

ershadul1 committed:
Modify user show template to show past and upcoming events separately

ershadul1 committed:
Add past and upcoming scopes in event model

ershadul1 committed:
Modify index view template to separately show past and upcoming events

ershadul1 committed:
Remove navigation links from index view template

ershadul1 committed:
Add navigation links in application layout

RNtaate committed:
Add sign in button to nav bar

RNtaate committed:
Update create method to register invited guests.

RNtaate committed:
Create event_attendance form partial

RNtaate committed:
Render event_attendance form partial in events new template

RNtaate committed:
Update destroy method in session to redirect to rootpath

RNtaate committed:
Remove logic from application view template.

RNtaate committed:
Add testing for presence, uniqueness validations for user model

RNtaate committed:
Add sign up button to sign_out partial

ershadul1 committed:
Add association tests for user model

ershadul1 committed:
Add associations tests for event model

ershadul1 committed:
Add request tests for event controller

RNtaate committed:
Add request testing for create method in sessions

RNtaate committed:
Add testing for show and new methods for users controller

ershadul1 committed:
Delete unwanted file

ershadul1 committed:
Add integration test for sign in process

ershadul1 committed:
Add integration test for sign up process

ershadul1 committed:
Add integration test for create new event process

RNtaate committed:
Fix rubocop offenses

RNtaate committed:
Add tests.yml to github work flows

ershadul1 committed:
Add pg gem for production

ershadul1 committed:
Fix rubocop offences

RNtaate committed:
Update README file

RNtaate committed:
Update testing of show, new, create methods for event_controller

RNtaate committed:
Update README.md

maelfosso commented on app/views/users/show.html.erb:
```diff
@@ -0,0 +1,13 @@
+<h1>Welcome <%= @user.name %> </h1>
+
+<h3>Previously attended events</h3>
+<% @user.attended_events.where("date < ?", Date.today).each do |event|%>
+  <%= link_to event.description, event_path(event.id) %><br>
+<% end %>
```
The better to do it it's by using Event scope
```rb
@user.attended_events.past
```
Create a variable in your action controller and use it here (in that view)

maelfosso commented on app/views/users/show.html.erb:
```diff
@@ -0,0 +1,13 @@
+<% end %>
+
+<h3>Upcoming events</h3>
+<% @user.attended_events.where("date >= ?", Date.today).each do |event|%>
+  <%= link_to event.description, event_path(event.id) %><br>
+<% end %>
```
The better to do it it's by using Event scope
```rb
@user.attended_events.upcoming
```
Create a variable in your action controller and use it here (in that view)
```rb
def show
  @upcoming = @user.attended_events.upcoming
  ...
  ...
end
```

maelfosso reviewed (changes requested):
# Changes required :red_circle: 

Hello,

Thank you for the job done so far.
Your application is working like a charm.
However, there are some missing requirements

1. You didn't write integrations tests, as requested [here](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png), for **events** and **users** management. Learn more about [here](https://hackernoon.com/how-to-build-awesome-integration-tests-with-capybara-j9333y68)

For the unit test, I suggest you use the **shoulda-matcher** gem. It's easy and fast.

2. It's only possible to attend an event during its creation. Please, add the possibility to attend the event after the event is created

3. As indicated [here](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png), the user show page must also display the event the user has created which is not the case in your application. Kindly fix that, please.

4. Read the comment left below

Kindly make the suggested changes and submit a new code review request
Thanks

Mael FOSSO

ershadul1 committed:
Transfer scopes from event.rb to application_record.rb

ershadul1 committed:
Use scopes to represent queries in view

ershadul1 committed:
Change rspec core version for github action

ershadul1 committed:
Remove unwanted file

ershadul1 committed:
Add dependent destroy in event and user model

ershadul1 committed:
Add edit and update method  in events controller

ershadul1 committed:
Add view template for event edit page

ershadul1 committed:
Add event edit link in event show page and create corresponding helper method

RNtaate committed:
Add styling to layout and view templates

RNtaate committed:
Add visibility size to collection_select in form partial

RNtaate committed:
Update html design in application, events new, users new view templates

RNtaate committed:
Update README screenshots

RNtaate committed:
Update README file

RNtaate committed:
Update authentication in events controller

RNtaate committed:
Add integration testing for edit event process

RNtaate committed:
Add selenium gem and install with bundler

RNtaate committed:
Configure capybara gem in rails, spec helper files

RNtaate committed:
Update integration testing for create, edit event, signin, sign up features

RNtaate committed:
Fix rubocop offenses

RNtaate commented:
## Changes made 
- [X] Added possibility to attend an event after the event is created.
- [X] Added created events to a user's show page.
- [X] Used Event scope to represent querries in view templates.
- [X] Made changes to integration tests as required by project specifications.

Dandush03 reviewed (approved):
# Status: :rocket: Project Approved :green_circle: 
### Great work, keep it up
## :tophat: Happy Coding :computer:
